main.js

```diff
@@ -30,40 +22,24 @@ class InitLayer {
 // lancement des web worker pour remonter les datas
 
 const giveData = new Worker("info.js");
+let federations = {};
+let infoCount = 0;
 
-let federations = [];
-let fedLayers = [];
-let compteurBdd = 0;
 giveData.onmessage = (informations) => {
-  // forEach date put one more in compteur, at the end we'll know if it's finish or not (cf last else)
-  compteurBdd++;
-  //information from webworker bitch !
+  infoCount++
   let [lat, lng, federation, city, length] = informations.data;
-
-  // looking for it's the first time for this federation, if yes so do ...
-  if (federations.indexOf(federation) === -1 ) {
-    // initiate a layerGroup with is first marker and an federation attribute to reconize it
-    layerGroup = new InitLayer(federation, lat, lng);
-    // increase layerGroup to iterate inside after
-    fedLayers.push(layerGroup);
-    // increase federation to iterate inside after
-    federations.push(federation);
-
-  // ... if not and the are again item in
-  } else if ( length !== compteurBdd ){
-    // looking for grab the good federation Layer item
-    for (var i = 0; i < fedLayers.length; i++) {
-      // when find do ...
-      if (fedLayers[i].federation === federation) {
-        // ... add marker into layerGroup according its federation
-        fedLayers[i].layer.addLayer(L.marker([lat, lng]));
-      }
-    }
+  // federation key is not registered yet
+  if (!(federation in federations)) {
```

Comment:
If the key 'BC' is not the the `federations` object, then declare `federations['BC']` as an empty list.